src/cpp/src/group_beam_searcher.cpp

```diff
@@ -412,8 +412,14 @@ EncodedResults beam_search(ov::InferRequest& lm,
         }
         size_t batch_size = next_tokens.size();
         // Set pointers
-        lm.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
-        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
+        ov::Tensor next_tokens_tensor(ov::element::i64, {batch_size, 1});
+        std::memcpy(next_tokens_tensor.data(), next_tokens.data(), next_tokens_tensor.get_byte_size());
+        lm.set_tensor("input_ids", next_tokens_tensor);
+
+        ov::Tensor next_beams_tensor(ov::element::i32, {batch_size});
+        std::memcpy(next_beams_tensor.data(), next_beams.data(), next_beams_tensor.get_byte_size());
```

Comment:
is it possible to reset `Tensor`s before exiting pipeline instead of copying data?

Reply:
done.

Reply:
Is it ok to make it only for beam search or should we reset inputs on pipeline level?